### ISSUE TITLE
Hero gets stuck after battles/item pick ups

### ISSUE THREAD

**Issue by hgdagon** [ihhub/fheroes2#1647]
Latest git build. After finishing seemingly any action with a popup window, the hero gets stuck. Cursor is an arrow, instead of a horse. Looking at the map, the destination "cross" is out of the map bounds. Happened twice during the same game, first time got fixed by skipping the day, second time it didn't work, I skipped a whole week, but I couldn't move.

![destination](https://user-images.githubusercontent.com/15173076/92996784-094c0b00-f4c3-11ea-8a6b-bbd77aa3c843.png)

**Comment by Branikolog** [comment]
Can you, please, check this issue on 0.8.1 release?
Can you also attach a savefile?

**Comment by ihhub** [comment]
Hi @idshibanov , could you please take a look at this issue? I suspect that this is due to our path finding changes.

**Comment by idshibanov** [comment]
I agree, most likely it's a regression after recent PR.

@hgdagon can you upload a savegame with bug? Couldn't reproduce it yet.

**Comment by hgdagon** [comment]
[Here](https://github.com/ihhub/fheroes2/files/5213595/save.zip)'s the save file. Although, it seems like I stirred up a commotion for no reason.  I still haven't been able to build this myself, so I used the AppVeyor artifact, and having checked it again, it's not the latest commit, it's [this](#1645 ) PR. For whatever reason, I assumed the latest build available there would not be a PR. My mistake.

**Comment by idshibanov** [comment]
@hgdagon Alright, what I see on your save is that your hero is stuck standing next to the pikeman monster stack. Combat should have happened but it doesn't trigger/at the same time game doesn't allow you to move.

This could be two bugs joining the forces to lock down hero: wrong impassibility set on the tile plus logic in ActionNewPosition. Other heroes are not affected and can path/move freely (AI heroes or if you recruit a new one). Will try to fix it in the near future, savegame helped for sure.

**Comment by hgdagon** [comment]
@idshibanov Do note that this is for a PR, not the main repo. I confused the 2, since they both mentioned the work `cursor`, I thought I was downloading the artifact for the latest commit of the main repo.

That said, in that save, the battle has already taken place, I closed the post-battle dialog and that's when I was unable to move. I don't remember if it was a battle or an item pick up the first time I encountered "the lock", but again it was after I closed a popup window.
